test/bigtable/filter.js

```diff
@@ -0,0 +1,677 @@
+      });
+    });
+
+    it('should only create an end bound', function() {
+      var end = 'b';
+      var key = 'Key';
+
+      var range = Filter.createRange(null, end, key);
+
+      assert.deepEqual(range, {
+        endKeyInclusive: end
+      });
+    });
+
+    it('should optionally accept inclusive flags', function() {
+      var start = {
+        value: 'a',
+        inclusive: false
+      };
+
+      var end = {
+        value: 'b',
+        inclusive: false
+      };
+
+      var key = 'Key';
+
+      var range = Filter.createRange(start, end, key);
+
+      assert.deepEqual(range, {
+        startKeyExclusive: start.value,
+        endKeyExclusive: end.value
+      });
+    });
+  });
+
+  describe('parse', function() {
+    it('should call each individual filter method', function() {
+      var row = Filter.prototype.row;
+      var value = Filter.prototype.value;
+
+      Filter.prototype.row = spy();
+      Filter.prototype.value = spy();
+
+      var fakeFilter = [{
+        row: 'a'
+      }, {
+        value: 'b'
+      }];
+
+      Filter.parse(fakeFilter);
+
+      assert.strictEqual(Filter.prototype.row.calls.length, 1);
+      assert.strictEqual(Filter.prototype.row.calls[0].args[0], 'a');
+      assert.strictEqual(Filter.prototype.value.calls.length, 1);
+      assert.strictEqual(Filter.prototype.value.calls[0].args[0], 'b');
+
+      Filter.prototype.row = row;
+      Filter.prototype.value = value;
+    });
+
+    it('should throw an error for unknown filters', function() {
+      var fakeFilter = [{
+        wat: 'a'
+      }];
+
+      assert.throws(
+        Filter.parse.bind(null, fakeFilter),
+        /Unknown filter\: wat/
+      );
+    });
+
+    it('should return the filter in JSON form', function() {
+      var toProto = Filter.prototype.toProto;
+      var fakeProto = { a: 'a' };
+
+      Filter.prototype.toProto = spy(function() {
+        return fakeProto;
+      });
+
+      var fakeFilter = [{
+        column: 'a'
+      }];
+
+      var parsedFilter = Filter.parse(fakeFilter);
+
+      assert.strictEqual(parsedFilter, fakeProto);
+      assert.strictEqual(Filter.prototype.toProto.calls.length, 1);
+
+      Filter.prototype.toProto = toProto;
+    });
+  });
+
+  describe('all', function() {
+    it('should create a pass all filter when set to true', function(done) {
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'passAllFilter');
+        assert.strictEqual(value, true);
+        done();
+      };
+
+      filter.all(true);
+    });
+
+    it('should create a block all filter when set to false', function(done) {
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'blockAllFilter');
+        assert.strictEqual(value, true);
+        done();
+      };
+
+      filter.all(false);
+    });
+  });
+
+  describe('column', function() {
+    it('should set the column qualifier regex filter', function(done) {
```

Comment:
Could you add a test that only makes sure a string `column` argument is converted and works appropriately? While it is indirectly covered in the first test, it would be easy for a future contributor to miss that purpose.